### PR TITLE
New Block: Add Query Total block for displaying total query results or ranges

### LINKED ISSUE #62423: New Block: Query Results Count Block
## What problem does this address?
Currently there is no block to display the total # of results from a query.

## What is your proposed solution?
Offer a new block to display totals, this would probably most commonly be used next to the pagination block. Here are some screenshots of how we use it on our site.

Used at top above the query loop:
Screenshot 2024-06-07 at 9 23 21 AM

Used at bottom underneath Pagination links:
Screenshot 2024-06-07 at 9 22 07 AM

### PR DESCRIPTION
**Pull request by sarthaknagoshe2002**
Closes: #62423

## What?
This PR adds a core/query-total block to display query-related information, such as the total number of results or the range of results currently displayed.

## Why?
Currently, there is no block that displays the total number of query results or the range of displayed results, limiting users' ability to convey query information effectively. This block enhances usability and customization options in the block editor.

## How?

- Block Features
  - Attributes: Includes `displayType` to toggle between "Total Results" and "Range Display."
  - Server-Side Rendering: Dynamic data fetched using `query context` (queryId, query).
  - Customization: Supports typography, color, spacing, and alignment options.
- Key Functionalities
  - "Total Results" mode: Displays the total number of query results.
  - "Range Display" mode: Shows the current range and total number of results.

## Testing Instructions

1. Add a Query Loop block to a post or page.
2. Insert the Query Total block inside or alongside the Query Loop.
3. Set the `displayType` attribute to "Total Results" or "Range Display" via block controls.
4. Preview and verify the output:
5. Ensure total results are displayed correctly in "Total Results" mode.
6. Verify range calculations in "Range Display" mode.
7. Test with various query contexts to ensure accurate dynamic rendering.

## Screenshots or screencast 

https://github.com/user-attachments/assets/f9497eab-552a-4586-b7c3-1c19ddda821c

### CODE REVIEW

**Comment by sarthaknagoshe2002** [comment]
I’m relatively new to contributing, but I’ve taken a bold step with this PR. I apologize in advance if I’ve made any mistakes along the way!  
I’m open to all kinds of feedback, especially since aspects like the block name, icon, and variation icon are open to discussion.  
If I’ve overlooked any steps related to adding a new block, please let me know—I noticed a test case failing for a backport PR 😅.  
I’m also working on the test cases and plan to include them either in this PR or, if acceptable, in a separate one.

File: packages/block-library/src/query-total/block.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{
+	"$schema": "https://schemas.wp.org/trunk/block.json",
+	"apiVersion": 2,
+	"name": "core/query-total",
+	"title": "Query Total",
+	"category": "theme",
+	"parent": [ "core/query-pagination", "core/query" ],
```
Comment (fabiankaegy) [inline packages/block-library/src/query-total/block.json]: I would suggest we change this to just have the query as it's ancestor. That way it can appear anywhere deeply nested in the query. 

```suggestion
	"ancestor": [ "core/query" ],
```

File: packages/block-library/src/query-total/edit.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	const [ totalResults, setTotalResults ] = useState( 0 );
+	const [ currentPage, setCurrentPage ] = useState( 1 );
+	const [ resultsPerPage, setResultsPerPage ] = useState( 10 );
+
+	// Fetch or calculate the total results, current page, and results per page.
+	useEffect( () => {
+		// Mock values for demonstration. Replace with actual query data.
+		const mockTotalResults = 12;
+		const mockCurrentPage = 1;
+		const mockResultsPerPage = 10;
```
Comment (fabiankaegy) [inline packages/block-library/src/query-total/edit.js]: Does this mean this PR should still be a draft?

File: packages/block-library/src/query-total/edit.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+	const calculateRange = () => {
+		const start = ( currentPage - 1 ) * resultsPerPage + 1;
+		const end = Math.min( currentPage * resultsPerPage, totalResults );
+		return `${ start } – ${ end }`;
+	};
+
+	// Block properties and classes.
+	const blockProps = useBlockProps( {
+		className: clsx( className, 'query-total-block' ),
+	} );
```
Comment (fabiankaegy) [inline packages/block-library/src/query-total/edit.js]: Why do we have the additional `query-total-block` class here? Can we not just use the `wp-block-query-total` class we already get out of the box? 

```suggestion
	const blockProps = useBlockProps();
```

File: packages/block-library/src/query-total/edit.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+					</p>
+				</PanelBody>
+			</InspectorControls>
+		</>
+	);
+
+	// Render output based on the selected display type.
+	const renderDisplay = () => {
+		if ( displayType === 'total-results' ) {
+			return <div>{ `${ totalResults } results found` }</div>;
```
Comment (fabiankaegy) [inline packages/block-library/src/query-total/edit.js]: The string here should be translatable

File: packages/block-library/src/query-total/edit.js [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+		}
+
+		if ( displayType === 'range-display' ) {
+			return (
+				<div>{ `Displaying ${ calculateRange() } of ${ totalResults }` }</div>
```
Comment (fabiankaegy) [inline packages/block-library/src/query-total/edit.js]: The string here should be translatable

Comment (fabiankaegy) [inline packages/block-library/src/query-total/edit.js]: I also wonder if there is any way in which we can make this text customizable? 🤔

**Review by fabiankaegy — COMMENTED** [review]
Having this in core is going to be super helpful! :) Added some initial notes 👍

File: packages/block-library/src/query-total/edit.js (hunk 2 above)
Comment (sarthaknagoshe2002) [reply packages/block-library/src/query-total/edit.js]: @fabiankaegy actually these mock variable will only be displayed in the editor similar to the dummy results of the query loop.

File: packages/block-library/src/query-total/edit.js (hunk 3 above)
Comment (sarthaknagoshe2002) [reply packages/block-library/src/query-total/edit.js]: umm apologies 🙌🏻, I just missed that part.

File: packages/block-library/src/query-total/edit.js (hunk 5 above)
Comment (sarthaknagoshe2002) [reply packages/block-library/src/query-total/edit.js]: Oh I am already working it on.
I thought that it would be better if I get this approved first & then add the customisable text & I was even planning on adding more variations. 😇😊

File: packages/block-library/src/query-total/edit.js (hunk 2 above)
Comment (fabiankaegy) [reply packages/block-library/src/query-total/edit.js]: In that case we don't need this `useEfect` and the `useState` instead can just set those static variables directly 🤔

Comment (sarthaknagoshe2002) [reply packages/block-library/src/query-total/edit.js]: Rectifying it right away🚀

**Comment by sarthaknagoshe2002** [comment]
@fabiankaegy I’ve implemented the requested changes.  

Could you please help me with the failing test? I’m currently working on the `fixture` to address the failing unit tests, but I’m unsure about how to proceed with the other failing tests. Any guidance would be greatly appreciated!

**Comment by fabiankaegy** [comment]
> @fabiankaegy I’ve implemented the requested changes.
> 
> Could you please help me with the failing test? I’m currently working on the `fixture` to address the failing unit tests, but I’m unsure about how to proceed with the other failing tests. Any guidance would be greatly appreciated!

I think all the JS ones are due to the missing fixture. 

The PHP ones are because the two functions you added are missing an `@since` tag in their doc comments https://github.com/WordPress/gutenberg/actions/runs/12182411187/job/33981359884?pr=67629

**Comment by sarthaknagoshe2002** [comment]
I noticed those error messages but i am unsure about the value of the `@since` tag

File: packages/block-library/src/query-total/index.php [hunk 6]
```diff
@@ -7,6 +7,8 @@
 
 /**
  * Renders the `query-total` block on the server.
+ * 
+ * @since 6.7.0
```
Comment (fabiankaegy) [inline packages/block-library/src/query-total/index.php]:
```suggestion
 * @since 6.8.0
```

File: packages/block-library/src/query-total/index.php [hunk 7]
```diff
@@ -70,6 +72,8 @@ function render_block_core_query_total( $attributes, $content, $block ) {
 
 /**
  * Registers the `query-total` block.
+ * 
+ * @since 6.7.0
```
Comment (fabiankaegy) [inline packages/block-library/src/query-total/index.php]:
```suggestion
 * @since 6.8.0
```

File: packages/block-library/src/query-total/index.php [hunk 8]
```diff
@@ -70,6 +72,8 @@ function render_block_core_query_total( $attributes, $content, $block ) {
 
 /**
  * Registers the `query-total` block.
+ *
```
Comment (fabiankaegy) [inline packages/block-library/src/query-total/index.php]:
```suggestion
 *
```

File: packages/block-library/src/query-total/index.php [hunk 9]
```diff
@@ -7,6 +7,8 @@
 
 /**
  * Renders the `query-total` block on the server.
+ *
```
Comment (fabiankaegy) [inline packages/block-library/src/query-total/index.php]:
```suggestion
 *
```

**Comment by gziolo** [comment]
It looks like there is also an experimental block Comments Count that would be very similar. I don't know why it is still experimental, but it was raised by @carolinan recently on WordPress Slack: https://wordpress.slack.com/archives/C02QB2JS7/p1733463619396909.

In case we decide to add this Query related block, we should mirror the handling with the one related to Comments.

**Comment by sarthaknagoshe2002** [comment]
> In case we decide to add this Query related block, we should mirror the handling with the one related to Comments.

@gziolo here, by handling do you mean the appearance??
Cuz as far as I know the internal logic is quite consistent with other blocks like `pagination numbers` block as it works on similar lines.

File: packages/block-library/src/query-total/block.json (hunk 1 above)
Comment (gziolo) [reply packages/block-library/src/query-total/block.json]: Good call 👍

File: packages/block-library/src/query-total/block.json [hunk 10]
```diff
@@ -0,0 +1,45 @@
+{
+	"$schema": "https://schemas.wp.org/trunk/block.json",
+	"apiVersion": 2,
```
Comment (gziolo) [inline packages/block-library/src/query-total/block.json]:
```suggestion
	"apiVersion": 3,
```

3 is the most up to date version.

File: packages/block-library/src/query-total/index.php [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+			$range_text = $start === $end
+				? sprintf( 'Displaying <strong>%d</strong> of <strong>%d</strong>', $start, $max_rows )
+				: sprintf( 'Displaying <strong>%d – %d</strong> of <strong>%d</strong>', $start, $end, $max_rows );
+
+			$output = sprintf( '<p>%s</p>', $range_text );
+			break;
+
+		default:
+			$output = sprintf(
+				'<p><strong>%d</strong> %s found</p>',
```
Comment (gziolo) [inline packages/block-library/src/query-total/index.php]: All these strings mixed with HTML code need to be translatable.

Comment (sarthaknagoshe2002) [reply packages/block-library/src/query-total/index.php]: This is fixed now 😇

Comment (gziolo) [reply packages/block-library/src/query-total/index.php]: I need to check similar code usage. Best practices are explained at: https://developer.wordpress.org/apis/internationalization/internationalization-guidelines/. Mixing HTML in the translation makes it all more difficult. Overall full phrases should be translatable rather than it’s chunks.

File: packages/block-library/src/query-total/edit.js [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+export default function QueryTotalEdit( { attributes, setAttributes } ) {
+	const { displayType } = attributes;
+
+	// Mock values.
+	const totalResults = 12;
+	const currentPage = 1;
+	const resultsPerPage = 10;
+
+	// Helper to calculate the range for "Displaying X – Y of Z".
+	const calculateRange = () => {
```
Comment (gziolo) [inline packages/block-library/src/query-total/edit.js]: This can be calculated once for the props used in the function and memoized with `useMemo` React hook.

File: packages/block-library/src/query-total/edit.js [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+
+/**
+ * Internal dependencies
+ */
+import { resultsFound, displayingResults } from './icons';
+
+export default function QueryTotalEdit( { attributes, setAttributes } ) {
+	const { displayType } = attributes;
+
+	// Mock values.
```
Comment (gziolo) [inline packages/block-library/src/query-total/edit.js]: What’s the plan with these mocked values? Should they be read from block context?

**Comment by gziolo** [comment]
> here, by handling do you mean the appearance??
> Cuz as far as I know the internal logic is quite consistent with other blocks like pagination numbers block as it works on similar lines.

I meant that it would be great to follow up next and improve Comments Count block once these changes are done. 

By the way, on the technical level this PR is close to ready. I will defer decisions to @fabiankaegy regarding how the user experience looks like.

File: packages/block-library/src/query-total/edit.js (hunk 13 above)
Comment (sarthaknagoshe2002) [reply packages/block-library/src/query-total/edit.js]: <s> If we look at the `page numbers` block, instead of querying the actual page numbers it shows the page numbers upto 10 only the `mid size` is changed.
technically, the are also mock values.
Similarly, the same idea was implemented here. 
</s>
Removed the usage of these mock value & hard coded the strings

File: packages/block-library/src/query-total/edit.js (hunk 12 above)
Comment (sarthaknagoshe2002) [reply packages/block-library/src/query-total/edit.js:30]: @gziolo I am thinking of getting rid of the function and completely hard coding the values for editor like this.

```js
if ( displayType === 'range-display' ) {
  return (
    <div>
      { __( 'Displaying 1 - 10 of 12' ) }
    </div>
  );
}
```

What are your views?

File: packages/block-library/src/query-total/index.php (hunk 11 above)
Comment (sarthaknagoshe2002) [reply packages/block-library/src/query-total/index.php]: I have modified the approach to ensure the entire phrase is translatable and not the chunks of it

File: packages/block-library/src/query-total/edit.js (hunk 12 above)
Comment (sarthaknagoshe2002) [reply packages/block-library/src/query-total/edit.js]: To simplify everything I have removed this function & hard coded the output on the editor side

**Comment by fabiankaegy** [comment]
From my perspective this also is pretty much ready to go from the editorial experience.

I would love to get @jameskoster / @jasmussen to take a look a the block icon as I think that is not an ideal fit yet 🤔

Also as a follow up I would love to have a way to customize the label / omit the label entirely :)

**Comment by sarthaknagoshe2002** [comment]
> Also as a follow up I would love to have a way to customize the label / omit the label entirely :)

Yes, as previously discussed, I will be addressing this in a separate pull request.

File: packages/block-library/src/query-total/index.php (hunk 11 above)
Comment (gziolo) [reply packages/block-library/src/query-total/index.php]: Nice one, thank you!

File: packages/block-library/src/query-total/edit.js (hunk 12 above)
Comment (gziolo) [reply packages/block-library/src/query-total/edit.js]: Makes sense 👍🏻

File: packages/block-library/src/query-total/edit.js [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+					</p>
+				</PanelBody>
+			</InspectorControls>
+		</>
+	);
+
+	// Render output based on the selected display type.
+	const renderDisplay = () => {
+		if ( displayType === 'total-results' ) {
+			return <div>{ __( '12 results found' ) }</div>;
```
Comment (gziolo) [inline packages/block-library/src/query-total/edit.js:67]: My last question is where the number 12 comes from. I'm not sure how much these display elements need to represent real data, but in some cases, it would be possible to show the actual number of results and the current settings for items per page.

**Comment by fabiankaegy** [comment]
CleanShot 2024-12-10 at 08 52 38@2x

@sarthaknagoshe2002 Actually, regarding the settings sidebar. Right now you are adding an empty panel in the sidebar. We should change that to maybe move the dropdown of options from the block toolbar into the sidebar here.

File: packages/block-library/src/query-total/edit.js (hunk 14 above)
Comment (fabiankaegy) [reply packages/block-library/src/query-total/edit.js:67]: @gziolo We have a precedence for this in core. The query pagination also doesn't show the real pagination but has a good default value that showcases the block in action 🤔

I think it makes sense here also as `1 - 1 out of 1` doesn't really make sense

**Comment by jasmussen** [comment]
Tricky one. In fact I think all pagination, including this smmary which is related to pagination, need a visual refresh. One instinct for making them related, is putting them in a box. For now, I drew a perhaps nicer summary icon, it's not a bad metaphor:

Screenshot 2024-12-10 at 08 52 58

```
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 4H6c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2Zm.5 14c0 .3-.2.5-.5.5H6c-.3 0-.5-.2-.5-.5V6c0-.3.2-.5.5-.5h12c.3 0 .5.2.5.5v12Zm-7-6-4.1 5h8.8v-3h-1.5v1.5h-4.2l2.9-3.5-2.9-3.5h4.2V10h1.5V7H7.4l4.1 5Z"/></svg>
```

If there's a better metaphor, let's go for it, but perhaps for now this can allow the feature to ship. And as noted, if we like it, let's follow up with better designs for the other pagination icons.

File: packages/block-library/src/query-total/edit.js (hunk 14 above)
Comment (gziolo) [reply packages/block-library/src/query-total/edit.js:67]: > I think it makes sense here also as 1 - 1 out of 1 doesn't really make sense

Now that you shared it. What should happen for edge cases?

- `1 -1 out of 1` - should it simply display `1` or an empty string?
- `1-5 out of 5` - should it display `1-5`?

Thank you for the explanation about the number `12`. It's now documented and we can proceed further 😄

Comment (gziolo) [reply packages/block-library/src/query-total/edit.js:67]: I see that it's mostly covered already in PHP 😄

Comment (fabiankaegy) [reply packages/block-library/src/query-total/edit.js:67]: I think the proper answer would still be: 

1 - 1 out of 1

The format should be:

`index of first item on this page` - `index of last item on this page` out of `total number of results`

File: packages/block-library/src/query-total/index.php [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+					'<strong>' . $end . '</strong>',
+					'<strong>' . $max_rows . '</strong>'
+				);
+			}
+
+			$output = sprintf( '<p>%s</p>', $range_text );
+			break;
+
+		default:
+			$output = sprintf(
```
Comment (gziolo) [inline packages/block-library/src/query-total/index.php:64]: Is `total-results` exactly the same as the default?

File: packages/block-library/src/query-total/edit.js (hunk 14 above)
Comment (sarthaknagoshe2002) [reply packages/block-library/src/query-total/edit.js:67]: Currently this is how it is handled on the frontend
Screenshot 2024-12-10 at 1 37 00 PM
Screenshot 2024-12-10 at 1 37 07 PM

**Comment by sarthaknagoshe2002** [comment]
> @sarthaknagoshe2002 Actually, regarding the settings sidebar. Right now you are adding an empty panel in the sidebar. We should change that to maybe move the dropdown of options from the block toolbar into the sidebar here.

For this I have tried to maintain consistency with the search block.

| Search block's Inspector Panel | Search block's Toolbar |
|-|-|
| Screenshot 2024-12-10 at 1 40 06 PM | Screenshot 2024-12-10 at 1 40 01 PM |

File: packages/block-library/src/query-total/edit.js (hunk 14 above)
Comment (gziolo) [reply packages/block-library/src/query-total/edit.js:67]: Looking at the logic, it will show `Displaying 1 of 1` which is perfectly fine.

File: packages/block-library/src/query-total/index.php (hunk 15 above)
Comment (sarthaknagoshe2002) [reply packages/block-library/src/query-total/index.php:64]: Umm yes and yes thats a redundancy there. It is intentional.

Having the `default case` and the `total-results` case use the same logic makes it clear that they are intended to produce the same output. It avoids any ambiguity about what happens when `displayType` is not provided or is set to `total-results`.

Also, if in future we decide to add or modify functionality for `total-results`, we can do so without impacting the `default case`.

To summarize from my perspective it just improves the readablity for future development

**Comment by sarthaknagoshe2002** [comment]
> Tricky one. In fact I think all pagination, including this smmary which is related to pagination, need a visual refresh. One instinct for making them related, is putting them in a box. For now, I drew a perhaps nicer summary icon, it's not a bad metaphor:
> 
> Screenshot 2024-12-10 at 08 52 58
> 
> ```
> <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 4H6c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2Zm.5 14c0 .3-.2.5-.5.5H6c-.3 0-.5-.2-.5-.5V6c0-.3.2-.5.5-.5h12c.3 0 .5.2.5.5v12Zm-7-6-4.1 5h8.8v-3h-1.5v1.5h-4.2l2.9-3.5-2.9-3.5h4.2V10h1.5V7H7.4l4.1 5Z"/></svg>
> ```
> 
> If there's a better metaphor, let's go for it, but perhaps for now this can allow the feature to ship. And as noted, if we like it, let's follow up with better designs for the other pagination icons.

This looks cool!

Screenshot 2024-12-10 at 1 45 30 PM

File: packages/block-library/src/query-total/index.php (hunk 15 above)
Comment (gziolo) [reply packages/block-library/src/query-total/index.php:64]: You can achieve the same effect with:
```php
case 'total-results':
default:
```

Besides, the attribute defaults to `total-results`, so there won't be a case where this value is undefined. Let's avoid the unnecessary duplication of code by focusing on what's possible today.

**Comment by fabiankaegy** [comment]
> > @sarthaknagoshe2002 Actually, regarding the settings sidebar. Right now you are adding an empty panel in the sidebar. We should change that to maybe move the dropdown of options from the block toolbar into the sidebar here.
> 
> For this I have tried to maintain consistency with the search block.
> 
> Search block's Inspector Panel	Search block's Toolbar
> Screenshot 2024-12-10 at 1 40 06 PM	Screenshot 2024-12-10 at 1 40 01 PM

The main thing here if we want to keep the control in the toolbar (which I'm fine with) is that we then should remove everything in the Inspector 🤔 Right now the inspector has some text in it but no controls

**Comment by sarthaknagoshe2002** [comment]
> The main thing here if we want to keep the control in the toolbar (which I'm fine with) is that we then should remove everything in the Inspector 🤔 Right now the inspector has some text in it but no controls

Yes, this makes the toolbar attract the user's attention, no distraction from the Inspector Panel.


Screenshot 2024-12-10 at 2 01 34 PM

File: packages/block-library/src/query-total/index.php (hunk 15 above)
Comment (sarthaknagoshe2002) [reply packages/block-library/src/query-total/index.php:64]: Fixed this 😇

File: packages/block-library/src/query-total/index.php [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+
+	// Calculate the range of posts being displayed.
+	$start = ( $current_page - 1 ) * $posts_per_page + 1;
+	$end   = min( $start + $posts_per_page - 1, $max_rows );
+
+	// Prepare the display based on the `displayType` attribute.
+	$output = '';
+	switch ( $attributes['displayType'] ) {
+		case 'total-results':
+		default:
```
Comment (gziolo) [inline packages/block-library/src/query-total/index.php]: The case for `range-display` needs to be moved before the `default` clause as it will never be reachable otherwise 😄

Comment (sarthaknagoshe2002) [reply packages/block-library/src/query-total/index.php]: I missed that in all the excitement! 😆

**Review by gziolo — APPROVED** [review]
As shared earlier, I'll approve the code aspects of the new block. It's looking solid from my perspective. Excellent collaboration @sarthaknagoshe2002. That was a fast and efficient feedback loop 🙇🏻 

Let's finalize the design decisions before proceeding with merging.

**Review by fabiankaegy — APPROVED** [review]
This works well and the experience in the editor is as expected :) 

Great job here! This is ready to go :)

**Comment by sarthaknagoshe2002** [comment]
@fabiankaegy @gziolo Thank you so much for your invaluable support with this PR. Honestly, this feels like a significant milestone in my journey as a contributor. I truly appreciate you guiding me and making me feel at ease as a rookie.

Thanks again for carrying me through and creating such a welcoming and supportive environment!

File: packages/block-library/src/query-total/index.php [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+
+			$output = sprintf( '<p>%s</p>', $range_text );
+			break;
+
+		case 'total-results':
+		default:
+			$output = sprintf(
+				'<p><strong>%d</strong> %s</p>',
+				$max_rows,
+				_n( 'result found', 'results found', $max_rows )
```
Comment (swissspidy) [inline packages/block-library/src/query-total/index.php:67]: Can we please fix this to not use string concatenation?

This would be better as `_n( '<strong>%d</strong> result found',  '<strong>%d</strong> results found', $max_rows )`

Noticed while reviewing #67994

@sarthaknagoshe2002 @fabiankaegy

Comment (gziolo) [reply packages/block-library/src/query-total/index.php:67]: Good one, @swissspidy. Thank you for flagging it. There were a few iterations on translations in this PR as documented in https://github.com/WordPress/gutenberg/pull/67629#discussion_r1876170366. This needs a bit of practice to get it right 😄

Comment (sarthaknagoshe2002) [reply packages/block-library/src/query-total/index.php:67]: I also think we should avoid enforcing styling on the displayed text, so removing the `<strong>` tag sounds like a good idea. What’s your take on this? @gziolo @fabiankaegy @swissspidy

Comment (sarthaknagoshe2002) [reply packages/block-library/src/query-total/index.php:67]: > This would be better as _n( '<strong>%d</strong> result found',  '<strong>%d</strong> results found', $max_rows )

Considering this https://github.com/WordPress/gutenberg/pull/67629#discussion_r1876278429 I believe it was a better option to use `_n( 'result found', 'results found', $max_rows )`

@swissspidy

Comment (swissspidy) [reply packages/block-library/src/query-total/index.php:67]: Well no, HTML is discouraged unless necessary.

Here, HTML is better than string concatenation. So it should be fixed.

Comment (sarthaknagoshe2002) [reply packages/block-library/src/query-total/index.php:67]: So after adopting the suggested changes & avoiding `<strong>` tag this is how things are looking.

```php
$output = sprintf(
  '<p>%s</p>',
  $max_rows,
  _n( '%d result found', '%d results found', $max_rows )
);
```

Does this require any changes?

Comment (swissspidy) [reply packages/block-library/src/query-total/index.php:67]: looks good to me, apart from the missing translator comment

Comment (sarthaknagoshe2002) [reply packages/block-library/src/query-total/index.php:67]: > looks good to me, apart from the missing translator comment

Yes, thats added.

Comment (sarthaknagoshe2002) [reply packages/block-library/src/query-total/index.php:67]: Umm there was a logical issue with the previous solution, this is the updated one

```php
// translators: %d: Total number of posts.
$total_text = sprintf( _n( '%d result found', '%d results found', $max_rows ), $max_rows );
$output = sprintf(
  '<p>%s</p>',
  $total_text,
);
```

Comment (sarthaknagoshe2002) [reply packages/block-library/src/query-total/index.php:67]: Added these changes through https://github.com/WordPress/gutenberg/pull/68150

Comment (sarthaknagoshe2002) [reply packages/block-library/src/query-total/index.php:67]: Can someone review & get this PR (https://github.com/WordPress/gutenberg/pull/68150) merged?
